src/DssSpell.sol

```diff
@@ -16,25 +16,108 @@
+import { SkyInstance } from "./dependencies/03-sky/SkyInstance.sol";
+
+import { UniV2PoolMigratorInit } from "./dependencies/04-univ2-pool-migrator/UniV2PoolMigratorInit.sol";
+
+interface PauseLike {
+    function delay() external view returns (uint256);
+    function plot(address, bytes32, bytes calldata, uint256) external;
+    function exec(address, bytes32, bytes calldata, uint256) external returns (bytes memory);
+}
+
+interface ChainlogLike {
+    function getAddress(bytes32) external view returns (address);
+}
+
+interface SpellActionLike {
+    function officeHours() external view returns (bool);
+    function description() external view returns (string memory);
+    function nextCastTime(uint256) external view returns (uint256);
+}
+
+contract DssExec {
+    ChainlogLike  constant public chainlog = ChainlogLike(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
+    uint256                public eta;
+    bytes                  public sig;
+    bool                   public done;
+    bytes32      immutable public tag;
+    address      immutable public action;
+    uint256      immutable public expiration;
+    PauseLike    immutable public pause;
+
+    uint256 constant public MIN_ETA = 1726574400; // 2024-09-17T12:00:00Z
+
+    // Provides a descriptive tag for bot consumption
+    // This should be modified weekly to provide a summary of the actions
+    // Hash: seth keccak -- "$(wget https://<executive-vote-canonical-post> -q -O - 2>/dev/null)"
+    function description() external view returns (string memory) {
+        return SpellActionLike(action).description();
+    }
+
+    function officeHours() external view returns (bool) {
+        return SpellActionLike(action).officeHours();
+    }
+
+    function nextCastTime() external view returns (uint256 castTime) {
+        return SpellActionLike(action).nextCastTime(eta);
+    }
+
+    // @param _description  A string description of the spell
+    // @param _expiration   The timestamp this spell will expire. (Ex. block.timestamp + 30 days)
+    // @param _spellAction  The address of the spell action
+    constructor(uint256 _expiration, address _spellAction) {
+        pause       = PauseLike(chainlog.getAddress("MCD_PAUSE"));
+        expiration  = _expiration;
+        action      = _spellAction;
+
+        sig = abi.encodeWithSignature("execute()");
+        bytes32 _tag;                    // Required for assembly access
+        address _action = _spellAction;  // Required for assembly access
+        assembly { _tag := extcodehash(_action) }
+        tag = _tag;
+    }
+
+    function schedule() public {
+        require(block.timestamp <= expiration, "This contract has expired");
+        require(eta == 0, "This spell has already been scheduled");
+        // Even if the spell gathers enough support too fast, we do not allow it to be executed before MIN_ETA.
+        eta = _max(block.timestamp + PauseLike(pause).delay(), MIN_ETA);
```

Comment:
I would suggest to: 
- Add the [original instruction](https://docs.google.com/spreadsheets/d/1w_z5WpqxzwreCcaveB2Ye1PP5B8QAHDglzyxKHG3CHw/edit?gid=1165645248#gid=1165645248&range=B30) on top
- Add additional `Note` to explain `_max` logic for the external readers, e.g.:
```
Note: in case the spell is scheduled later than planned, we have to switch back to the regular logic to respect GSM delay enforced here https://github.com/dapphub/ds-pause/blob/0763eafcf926fd2e073aee5f047f3decb842231c/src/pause.sol#L82
```